File: 1-two-sum/two-sum.java
class Solution {
     public int[] twoSum(int[] nums, int target) {

        Map<Integer, Integer> hashMap = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int value = target -nums[i];
            if (hashMap.containsKey(value)){
                return new int[]{hashMap.get(value),i};
            }
            hashMap.put(nums[i], i);
        }
        return null;
    }
}

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        for (int i = 0; i < nums.size(); i++){
            for (int j = 0; j < nums.size(); j++){
                if (i == j) continue;
                int temp = nums[j] + nums[i];
                if(temp == target){
                    result.push_back(i);
                    result.push_back(j);
                    break;
                }
            }
            if (result.size() > 1) break;
        }
        return result;
    }   
};